src/lj_arch.h

```diff
@@ -615,4 +615,13 @@ extern void *LJ_WIN_LOADLIBA(const char *path);
 #define LJ_52			0
 #endif
 
+/* Optimized string hashing, added by OpenResty. */
+#if (LUAJIT_TARGET == LUAJIT_ARCH_X64 && defined(__SSE4_2__) || LUAJIT_TARGET == LUAJIT_ARCH_ARM64 && __ARM_FEATURE_CRC32) && defined(__GNUC__)
```

Comment:
You shouldn't need the  `__SSE4_2__` and `__ARM_FEATURE_CRC32` checks at all.  `lj_str_hash.c` should always be built with `-msse4.2` and `-march=armv8-a+crc` for x86 and aarch64 respectively .  #75 has the necessary Makefile hacks to make that happen for sse4.2, you just need to replicate it for arm64.

Reply:
This is voluntary for the time being. Answered in my below comment on the PR's thread.